### ISSUE TITLE
SMTP protocol synchronization error (input sent without waiting for greeting): rejected connection from H=localhost

### ISSUE THREAD

**Issue by yeupou** [owncloud/mail#572]
Hello,

When trying to connect, I get "Error connecting to SMTP server" and the following in the exim logs

2015-05-02 17:29:14 SMTP protocol synchronization error (input sent without waiting for greeting): rejected connection from H=localhost [127.0.0.1] input="\026\003\001\002"
2015-05-02 17:29:27 SMTP protocol synchronization error (input sent without waiting for greeting): rejected connection from H=localhost [127.0.0.1] input="\026\003\001\002"
2015-05-02 17:29:51 SMTP protocol synchronization error (input sent without waiting for greeting): rejected connection from H=localhost [127.0.0.1] input="\026\003\001\002"

no matter what I set as SMTP conf. I want to use the localhost SMTP so there should not be much to configure. The SMTP daemon is already up and running and accept to send mail from local.

**Comment by Xenopathic** [comment]
It sounds like you have set the wrong security mode - try changing SSL/TLS to None or to STARTTLS. Those bytes you are seeing are the client handshake of an SSL negotiation, which clearly your server doesn't use.

**Comment by yeupou** [comment]
Ok, so it turns out that even when you put "none", it still tries to do some SSL and put as port number 587 (odd). 
When putting the regular port 25, then it works.

Aside from that, on a server that actually does support SSL/TLS, it should work nonetheless.

**Comment by Xenopathic** [comment]
So just to confirm - you tried None with a port of 587, and got these errors, but changing the port to 25 works fine?

**Comment by yeupou** [comment]
Yes.

**Comment by wurstchristoph** [comment]
@Xenopathic I just checked it - by default '587' and 'None' are selected. This is somehow strange because the default SMTP port is 25. I think we should set the default to '25' + 'None', or '587' + 'SSL' if we want to prefer connections over SSL

**Comment by Xenopathic** [comment]
@wurstchristoph Nope, 587 is the SMTP submission port, which should be used for any submissions. 25 is the Mail Transfer Agent (MTA) port, which is used for inter-server mail transfer. 465 is the SSL SMTP submission port, selected by default with the 'SSL/TLS' security option.

**Comment by TheGuv** [comment]
We have a  5-second delay before sending the SMTP greeting and deny any sender that tries to continue the transaction before the greeting has been sent (which is against the SMTP spec). This catches a great many spam bots. This mail app doesn't wait for the greeting and so gets denied. I've worked around it with a separate port just for this app, but IMHO this needs tweaking.

**Comment by DeepDiver1975** [comment]
> This mail app doesn't wait for the greeting and so gets denied.

This is basically an issue which should be addressed at the upstream project horde - please file an issue there - THX a lot!

**Comment by TheGuv** [comment]
wilco!

**Comment by webmarka** [comment]
Same problem on my side and unable to create an initial account in mail. But I see that the test email using the same email address with SMTP is working. Any progress on that ?
